src/screens/Home.tsx

```diff
@@ -3,152 +3,153 @@ import React, { ReactElement, useEffect, useState } from 'react';
 import DropdownMenu from '../components/DropdownMenu';
 import MapContainer from '../components/MapContainer';
 import PromotionList from '../components/PromotionList';
-import { usePromotionsList } from '../contexts/PromotionsListContext';
-import * as Promotion from '../types/promotion';
+import { DispatchAction, usePromotionsList } from '../contexts/PromotionsListContext';
+import { getEnum } from '../services/EnumService';
 import { Dropdown, DropdownType } from '../types/dropdown';
+import { Sort } from '../types/promotion';
+import Routes from '../utils/routes';
 
 const mapWidth = 60;
 
 export default function Home(): ReactElement {
   const [height, setHeight] = useState<string>('');
 
+  /* Options for each dropdown */
+  const [cuisineTypes, setCuisineTypes] = useState<string[]>([]);
+  const [daysOfWeek, setDaysOfWeek] = useState<string[]>([]);
+  const [discountTypes, setDiscountTypes] = useState<string[]>([]);
+  const [promotionTypes, setPromotionTypes] = useState<string[]>([]);
+
   const { dispatch } = usePromotionsList();
 
+  /**
+   * On component mount, load dropdown options
+   */
+  useEffect(() => {
+    const endpoints = [
+      {
+        endpoint: Routes.ENUMS.CUISINE_TYPES,
+        setOptions: setCuisineTypes,
+      },
+      {
+        endpoint: Routes.ENUMS.DAYS_OF_WEEK,
+        setOptions: setDaysOfWeek,
+      },
+      {
+        endpoint: Routes.ENUMS.DISCOUNT_TYPES,
+        setOptions: setDiscountTypes,
+      },
+      {
+        endpoint: Routes.ENUMS.PROMOTION_TYPES,
+        setOptions: setPromotionTypes,
+      },
+    ];
+    endpoints.forEach(({ endpoint, setOptions }) => {
+      getEnum(endpoint)
+        .then((options) => setOptions(options))
+        .catch(() => setOptions([]));
+    });
```

Comment:
Would these endpoints be called each time we refresh the page? Wondering if it would be more efficient if you just have an `/enums` endpoint you could call, which would return all the enums and their values. 

```
{
  PROMOTION_TYPES: [...],
  DISCOUNT_TYPES: [...],
  etc.
}
```

Reply:
Ohh like an `/enums` endpoint on the BE? Actually, this might be helpful because you're right these would be called every time we refresh the page unless we persisted these to local storage -- and if we do that we'd have to implement some periodic invalidation of the local storage... What do you think?

Reply:
Yeah so that way if we add more enums, you don't need to make as many changes here and it can all be in one http request. Not sure if that's common practice, but I think it would be beneficial. Yeah I think we can do an `/enums` endpoint in the BE, and then potentailly worry about local storage afterwards